src/xdp/trn_transit_xdp.c

```diff
@@ -509,8 +448,30 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 
 	// todo: add conn_track related logic properly
 	if (pkt->inner_ipv4_tuple.protocol == IPPROTO_TCP || pkt->inner_ipv4_tuple.protocol == IPPROTO_UDP) {
-		if (conntrack_is_reply_of_tracked_conn(&conn_track_cache, tunnel_id, &pkt->inner_ipv4_tuple))
+		// todo: handle udp reply even when the originated connection is marked as denied;
+		// this is necessary to check the derived policy rules as policy night have been updated to allow;
+		// the impl will be after the conn_track has connection states.
+		if (conntrack_is_reply_of_tracked_conn(&conn_track_cache, tunnel_id, &pkt->inner_ipv4_tuple)){
+			if (pkt->inner_ipv4_tuple.protocol == IPPROTO_UDP) {
+				struct ipv4_tuple_t originated_tuple = {
+					.protocol = IPPROTO_UDP,
+					.saddr = pkt->inner_ipv4_tuple.daddr,
+					.daddr = pkt->inner_ipv4_tuple.saddr,
+					.sport = pkt->inner_ipv4_tuple.dport,
+					.dport = pkt->inner_ipv4_tuple.sport,
+				};
+				if (0 != egress_policy_check(tunnel_id, &originated_tuple)){
+					bpf_debug("[Transit:vpc 0x%lx] ABORTED: packet to 0x%x from 0x%x ingress denied, reply of a denied UDP conn\n",
+						tunnel_id,
+						bpf_ntohl(pkt->inner_ipv4_tuple.daddr),
+						bpf_ntohl(pkt->inner_ipv4_tuple.saddr));
+					// todo: to change conn state if it is available instead of removing it from the map
+					conntrack_remove_tcpudp_conn(&conn_track_cache, tunnel_id, &originated_tuple);
+					return XDP_ABORTED;
+				}
+			}
 			goto xdp_continue;
```

Comment:
Can we avoid goto?  (can be done in later PRs, sure)

Reply:
refactor is planned; will get rid of goto in next few PR soon.